### ISSUE TITLE
Infer fails on Presto

### ISSUE THREAD

**Issue by electrum** [facebook/infer#252]
https://github.com/facebook/presto

```
[dphillips@laptop:~/src/presto master] infer -- mvn clean compile
ERROR: failure during compilation command.
You can run the failing compilation command again by copy-pasting the
lines below in your terminal:

"""
python <<EOF
import subprocess
cmd = ['javac', '-g', '-cp', u'/usr/local/Cellar/infer/0.6.0/libexec/infer/lib/python/inferlib/../../../lib/java/processor.jar:/Users/dphillips/src/presto/presto-spi/target/classes:/Users/dphillips/.m2/repository/io/airlift/slice/0.19/slice-0.19.jar:/Users/dphillips/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.4.4/jackson-annotations-2.4.4.jar:/Users/dphillips/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.4.4/jackson-databind-2.4.4.jar:/Users/dphillips/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.4.4/jackson-core-2.4.4.jar:/Users/dphillips/.m2/repository/org/openjdk/jol/jol-core/0.2/jol-core-0.2.jar:', '-d', '/Users/dphillips/src/presto/presto-spi/target/classes', '-sourcepath', '/Users/dphillips/src/presto/presto-spi/src/main/java:', '-s', '/Users/dphillips/src/presto/presto-spi/target/generated-sources/annotations', '-g', '-target', '1.8', '-source', '1.8', '-encoding', 'UTF-8', '-J-Duser.language=en', u'-AclassSourceMapOutputFilename=/var/folders/r6/q6h5x06x1wxd6kdq8g3twzrs0twt0q/T/classSourceMap_tOUmCd.json']
subprocess.check_call(cmd)
EOF
"""

javac: no source files
Usage: javac <options> <source files>
use -help for a list of possible options
Traceback (most recent call last):
  File "/usr/local/bin/infer", line 180, in <module>
    main()
  File "/usr/local/bin/infer", line 145, in main
    capture_exitcode = imported_module.gen_instance(args, cmd).capture()
  File "/usr/local/Cellar/infer/0.6.0/libexec/infer/lib/python/inferlib/capture/mvn.py", line 65, in capture
    return util.run_compilation_commands(cmds, clean_cmd)
  File "/usr/local/Cellar/infer/0.6.0/libexec/infer/lib/python/inferlib/capture/util.py", line 45, in run_compilation_commands
    if cmd.start() != os.EX_OK:
  File "/usr/local/Cellar/infer/0.6.0/libexec/infer/lib/python/inferlib/analyze.py", line 611, in start
    self.compile()
  File "/usr/local/Cellar/infer/0.6.0/libexec/infer/lib/python/inferlib/analyze.py", line 365, in compile
    return self.javac.run()
  File "/usr/local/Cellar/infer/0.6.0/libexec/infer/lib/python/inferlib/jwlib.py", line 113, in run
    subprocess.check_call(failing_cmd)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['javac', '-g', '-cp', u'/usr/local/Cellar/infer/0.6.0/libexec/infer/lib/python/inferlib/../../../lib/java/processor.jar:/Users/dphillips/src/presto/presto-spi/target/classes:/Users/dphillips/.m2/repository/io/airlift/slice/0.19/slice-0.19.jar:/Users/dphillips/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.4.4/jackson-annotations-2.4.4.jar:/Users/dphillips/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.4.4/jackson-databind-2.4.4.jar:/Users/dphillips/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.4.4/jackson-core-2.4.4.jar:/Users/dphillips/.m2/repository/org/openjdk/jol/jol-core/0.2/jol-core-0.2.jar:', '-d', '/Users/dphillips/src/presto/presto-spi/target/classes', '-sourcepath', '/Users/dphillips/src/presto/presto-spi/src/main/java:', '-s', '/Users/dphillips/src/presto/presto-spi/target/generated-sources/annotations', '-g', '-target', '1.8', '-source', '1.8', '-encoding', 'UTF-8', '-J-Duser.language=en', u'-AclassSourceMapOutputFilename=/var/folders/r6/q6h5x06x1wxd6kdq8g3twzrs0twt0q/T/classSourceMap_tOUmCd.json']' returned non-zero exit status 2
```

Specific commit: https://github.com/facebook/presto/commit/cee2e6a523fc05d4af9ea48ac2715889e49ae5c4

**Comment by sblackshear** [comment]
I fixed the issue with Maven (82b1dd0572d0afcdd6924d93849d8ec54f1c1b1f). 

Unfortunately, Infer still fails on Preto for me with "too many open files". This is happening because the Java frontend we use keeps opening .class files, then crashing when it can't parse the Java 8 bytecodes (and not closing the file). We are reporting this issue upstream; I will let you know if we find a workaround in the meantime.

**Comment by jvillard** [comment]
I just tried with latest presto and infer and it worked!

https://github.com/prestodb/presto/commit/6f04eaac123c1380718eafdb501fb11b1a45653a
54fdab889ad3ad98b66f82fdbe16f8e5b8251da8
